Trigger a mergability check on ghstack prs

PaliC opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115944

Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

PaliC committed:
Trigger a mergability check on ghstack prs

Trigger a mergability check on ghstack prs

[ghstack-poisoned]

huydhn commented on .github/workflows/check_mergability_ghstack.yml:
```diff
@@ -0,0 +1,22 @@
+    types: [opened, synchronize, reopened, edited]
+
+jobs:
+  mergability-and-dependencies-check:
```
I think that this fits better into `lint.yml` workflow.  Having a new workflow is ok, but lint seems like a natural way to keep this job

izaitsevfb replied:
One issue with `lint.yml` is that lint is triggered on pushes into main, which is redundant for mergeability (and in fact won't work for the workflow that requires PR number).

izaitsevfb reviewed (approved):
From the code perspective looks good. 

Let's add concurrency to the workflow (see `lint.yml` for example) and test it before merging.

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

izaitsevfb commented on .github/workflows/check_mergability_ghstack.yml:
```diff
@@ -0,0 +1,26 @@
+name: Check mergability and dependencies for ghstack prs
```
Nit: typo mergability -> merg**e**ability

(here and below)

izaitsevfb commented:
Hm, I don't see the new workflow triggered.

PaliC committed:
Update check_mergability_ghstack.yml

PaliC committed:
Update check_mergability_ghstack.yml

PaliC committed:
Update check_mergability_ghstack.yml

PaliC committed:
Update check_mergability_ghstack.yml

PaliC committed:
Update check_mergability_ghstack.yml

izaitsevfb commented on .github/workflows/check_mergability_ghstack.yml:
```diff
@@ -0,0 +1,30 @@
+name: Check mergability and dependencies for ghstack prs
```
```suggestion
name: Check mergeability and dependencies for ghstack prs
```

izaitsevfb replied:
Oh, and filename perhaps?

PaliC committed:
Update .github/workflows/check_mergability_ghstack.yml

Co-authored-by: Ivan Zaitsev <108101595+izaitsevfb@users.noreply.github.com>

huydhn reviewed (approved):
LGTM!

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

izaitsevfb reviewed (approved):
Awesome, thank you! Sorry about the nits!

huydhn commented on .github/workflows/check_mergeability_ghstack.yml:
```diff
@@ -0,0 +1,30 @@
+    outputs:
+      regex-match: ${{ steps.regex-match.outputs.match }}
+    steps:
+      - uses: actions/checkout@v2
```
Look like there are some deprecation warning coming from these actions, let's update them if possible https://github.com/pytorch/pytorch/actions/runs/7227759242:

* https://github.com/actions/checkout/tree/v4
* A bit of concern that `actions-ecosystem/action-regex-match` is not actively maintained anymore https://github.com/actions-ecosystem/action-regex-match/issues/396

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

PaliC committed:
Update on "Trigger a mergability check on ghstack prs"


Works to solve https://github.com/pytorch/test-infra/issues/4816

In conjunction with https://github.com/pytorch/test-infra/pull/4823 this pr should make it such that all ghstack prs kick off a job which is a mergability check.

Test plan, once https://github.com/pytorch/test-infra/pull/4823 is merged, I'll resubmit this diff to make sure the workflow job triggers.

[ghstack-poisoned]

PaliC commented:
@pytorchbot merge
